File: android/app/src/main/java/com/vxin/app/data/model/Contacts.kt
package com.vxin.app.data.model

import kotlinx.serialization.Serializable

/** 好友 —— GET /api/users/contacts */
@Serializable
data class Contact(
    val id: String,
    val username: String = "",
    val avatar: String = "",
    val bio: String = "",
    val status: String = "",
    val wechat_id: String = "",
    val remark: String? = null,
) {
    val displayName: String get() = remark?.takeIf { it.isNotBlank() } ?: username
}

/** 搜索结果用户 —— GET /api/users/search */
@Serializable
data class SearchUser(
    val id: String,
    val username: String = "",
    val avatar: String = "",
    val bio: String = "",
    val wechat_id: String = "",
)

/** 收到的好友申请 —— GET /api/users/friend-requests */
@Serializable
data class FriendRequest(
    val id: String,
    val from_id: String = "",
    val message: String = "",
    val status: String = "",
    val created_at: Long = 0,
    val username: String = "",
    val avatar: String = "",
    val wechat_id: String = "",
)

@Serializable
data class FriendRequestBody(val toId: String, val message: String = "")

/** 已发送的好友申请（GET friend-requests/sent） */
@Serializable
data class SentRequest(
    val id: String,
    val status: String = "",        // pending | accepted | rejected
    val message: String = "",
    val created_at: Long = 0,
    val toId: String = "",
    val username: String = "",
    val avatar: String = "",
    val wechat_id: String = "",
)

@Serializable
data class RemarkBody(val remark: String)

@Serializable
data class BlockedUser(
    val id: String,
    val username: String = "",
    val avatar: String = "",
)

@Serializable
data class HandleRequestBody(val action: String)   // accept | reject

@Serializable
data class SendRequestResponse(val success: Boolean = false, val autoAccepted: Boolean = false)

@Serializable
data class CreatePrivateBody(val userId: String)

@Serializable
data class CreateGroupBody(val name: String, val memberIds: List<String>)

@Serializable
data class CreateConversationResponse(val conversationId: String, val groupNumber: String? = null)

File: ios/Vxin/Data/Models/Contacts.swift
import Foundation

/// 好友 —— GET /api/users/contacts
struct Contact: Decodable, Identifiable, Hashable {
    let id: String
    var username: String = ""
    var avatar: String = ""
    var bio: String = ""
    var status: String = ""
    var wechatId: String = ""
    var remark: String?
    // 特权账户专属：好友最后在线时间（Unix 秒）。非特权账户后端不返回，保持 nil。
    var lastOnlineAt: Double?

    var displayName: String { (remark?.isEmpty == false ? remark! : username) }

    enum CodingKeys: String, CodingKey {
        case id, username, avatar, bio, status, remark
        case wechatId = "wechat_id"
        case lastOnlineAt = "last_online_at"
    }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id = try c.decode(String.self, forKey: .id)
        username = (try? c.decode(String.self, forKey: .username)) ?? ""
        avatar = (try? c.decode(String.self, forKey: .avatar)) ?? ""
        bio = (try? c.decode(String.self, forKey: .bio)) ?? ""
        status = (try? c.decode(String.self, forKey: .status)) ?? ""
        wechatId = (try? c.decode(String.self, forKey: .wechatId)) ?? ""
        remark = try? c.decode(String.self, forKey: .remark)
        lastOnlineAt = try? c.decode(Double.self, forKey: .lastOnlineAt)
    }
}

/// 搜索结果用户 —— GET /api/users/search
struct SearchUser: Decodable, Identifiable, Hashable {
    let id: String
    var username: String = ""
    var avatar: String = ""
    var bio: String = ""
    var wechatId: String = ""

    enum CodingKeys: String, CodingKey {
        case id, username, avatar, bio
        case wechatId = "wechat_id"
    }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id = try c.decode(String.self, forKey: .id)
        username = (try? c.decode(String.self, forKey: .username)) ?? ""
        avatar = (try? c.decode(String.self, forKey: .avatar)) ?? ""
        bio = (try? c.decode(String.self, forKey: .bio)) ?? ""
        wechatId = (try? c.decode(String.self, forKey: .wechatId)) ?? ""
    }
}

/// 收到的好友申请 —— GET /api/users/friend-requests
struct FriendRequest: Decodable, Identifiable, Hashable {
    let id: String
    var fromId: String = ""
    var message: String = ""
    var status: String = ""
    var createdAt: Double = 0
    var username: String = ""
    var avatar: String = ""
    var wechatId: String = ""

    enum CodingKeys: String, CodingKey {
        case id, message, status, username, avatar
        case fromId = "from_id"
        case createdAt = "created_at"
        case wechatId = "wechat_id"
    }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id = try c.decode(String.self, forKey: .id)
        fromId = (try? c.decode(String.self, forKey: .fromId)) ?? ""
        message = (try? c.decode(String.self, forKey: .message)) ?? ""
        status = (try? c.decode(String.self, forKey: .status)) ?? ""
        createdAt = (try? c.decode(Double.self, forKey: .createdAt)) ?? 0
        username = (try? c.decode(String.self, forKey: .username)) ?? ""
        avatar = (try? c.decode(String.self, forKey: .avatar)) ?? ""
        wechatId = (try? c.decode(String.self, forKey: .wechatId)) ?? ""
    }
}
